sdk/servicebus/Azure.Messaging.ServiceBus/samples/TopicFilters/Program.cs

```diff
@@ -0,0 +1,242 @@
+
+        private static ServiceBusClient s_client;
+        private static ServiceBusAdministrationClient s_adminClient;
+        private static ServiceBusSender s_sender;
+
+        public static async Task Main(string[] args)
+        {
+            var command = new RootCommand("Demonstrates the Topic Filters feature of Azure Service Bus.")
+            {
+                new Option<string>(
+                    alias: "--namespace",
+                    description: "Fully qualified Service Bus Queue namespace to use") {Name = "FullyQualifiedNamespace"},
+                new Option<string>(
+                    alias: "--connection-variable",
+                    description: "The name of an environment variable containing the connection string to use.") {Name = "Connection"},
+            };
+            command.Handler = CommandHandler.Create<string, string>(RunAsync);
+            await command.InvokeAsync(args);
+        }
+
+        private static async Task RunAsync(string fullyQualifiedNamespace, string connection)
+        {
+            if (!string.IsNullOrEmpty(connection))
+            {
+                s_client = new ServiceBusClient(Environment.GetEnvironmentVariable(connection));
+                s_adminClient = new ServiceBusAdministrationClient(Environment.GetEnvironmentVariable(connection));
+            }
+            else if (!string.IsNullOrEmpty(fullyQualifiedNamespace))
+            {
+                var defaultAzureCredential = new DefaultAzureCredential();
+                s_client = new ServiceBusClient(fullyQualifiedNamespace, defaultAzureCredential);
+                s_adminClient = new ServiceBusAdministrationClient(fullyQualifiedNamespace, defaultAzureCredential);
+            }
+            else
+            {
+                throw new ArgumentException(
+                    "Either a fully qualified namespace or a connection string environment variable must be specified.");
+            }
+
+            Console.WriteLine($"Creating topic {TopicName}");
+            await s_adminClient.CreateTopicAsync(TopicName);
+
+            s_sender = s_client.CreateSender(TopicName);
+
+            // First Subscription is already created with default rule. Leave as is.
+            Console.WriteLine($"Creating subscription {NoFilterSubscriptionName}");
+            await s_adminClient.CreateSubscriptionAsync(TopicName, NoFilterSubscriptionName);
+
+            Console.WriteLine($"SubscriptionName: {NoFilterSubscriptionName}, Removing and re-adding Default Rule");
+            await s_adminClient.DeleteRuleAsync(TopicName, NoFilterSubscriptionName, RuleProperties.DefaultRuleName);
+            await s_adminClient.CreateRuleAsync(TopicName, NoFilterSubscriptionName,
+                new CreateRuleOptions(RuleProperties.DefaultRuleName, new TrueRuleFilter()));
+
+            // 2nd Subscription: Add SqlFilter on Subscription 2
```

Comment:
What's a `SqlFilter` and what's its significance?   What are we filtering on?   Why are we removing the default rule?

Reply:
A link to the documentation for the SQL Rule syntax would also be super helpful for folks here.

Reply:
We link to the docs in the README, but I can add links in the comments as well. The comments (and code) was more or less copied as-is from https://github.com/Azure/azure-service-bus/tree/master/samples/DotNet/GettingStarted/Microsoft.Azure.ServiceBus/TopicSubscriptionWithRuleOperationsSample

Reply:
Even just a mention of something like "go see the README if you want to know stuffs"  would be good by me.  When we had code-forward samples initially, we saw questions coming up in issues that we had answered in the README or elsewhere; people seemed to be jumping to the code and skipping the docs.